src/index.ts

```diff
@@ -0,0 +1,121 @@
+import { toUnicode } from 'punycode'
+
+import {
+    allowed,
+    combiningDiacriticException,
+    cyrillic,
+    cyrillicLikeLatin,
+    dangerous,
+    decimalDigitNumber,
+    deviation,
+    getScripts,
+    inclusion,
+    kanaCharacterException,
+    latinGreekCyrillicAscii,
+    nonAsciiLatin,
+    pureAscii,
+    removed
+} from './regexes'
+
+import { Validator as Interface } from './interfaces'
+
+class UrlFormatter {
+    public readonly url: string
+    public readonly isTldAscii: boolean
+    constructor(url: string) {
+        this.url = url;
+        this.isTldAscii = !this.url.substring(this.url.lastIndexOf('.')).startsWith('.xn--')
+    }
+    public labels(): string[] {
+        return this.url.split('.').map(label => toUnicode(label))
+    }
+}
+
+export class Validator implements Interface {

```

Comment:
I think it'd make a more ergonomic API if a single function was exported instead, so on import you could do
```ts
import {validate} from 'ens-validation';
validate('mytext')
```

Reply:
Either exporting this as a singleton or breaking the class down into functions would work!
```ts
export const validate = (new Validator()).validate
// or
// break down into multiple functions, and just export validate
```

Reply:
I think the latter option would be better, since you could then unit test each function separately to get better code coverage and modularity in tests (unit test + integration tests)

Reply:
Agreed, lets go with option 2.